### DISCUSSION TITLE
Month & year navigation behavior in standalone date picker with multiple months display

### DISCUSSION THREAD

**Discussion by smarlhens** [chakra-ui/ark#2941] [Q&A] [answered]
Hi there! 👋🏻

I hope you’re doing well. I have a question regarding the date picker component in `@ark-ui/vue` version `3.14.0`, specifically related to the `num-of-months` setting.

When `num-of-months` is set to **2**, I’ve noticed that while using the "Prev" and "Next" buttons in day view behaves as expected, there seems to be an inconsistency in month and year views. For example, when I’m in September 2024 and click "Prev," I would expect to navigate to September 2023 in month view, but I instead go to August 2023. In year view, I’m directed to August 2014, which seems unexpected.

Is this the intended behavior?

[Screencast from 2024-09-30 10-44-41.webm](https://github.com/user-attachments/assets/7187e902-08c2-4559-b754-570f279083bf)

[Repro repo](https://stackblitz.com/edit/vue-ark-ui-v3-v4nmfh?file=src%2Fcomponents%2Fdate-picker.vue)

Thank you for your assistance, and I appreciate all the work you do on this library!

Wishing you a great day! ☮️

**Comment by segunadebayo** [comment]
That's interesting. Our design assumption with the date picker was that the view switcher is only helpful for a single-month calendar.

We'll think through this and consider improving it to support this use-case

**Comment by segunadebayo** [comment, marked as answer]
We've revisited this and decided that for multi-month views, the recommended approach is to use the month and year select dropdowns instead of the prev/next view navigation:

```tsx
<DatePicker.Root numOfMonths={2}>
  <DatePicker.Control>
    <DatePicker.Input />
    <DatePicker.Trigger />
  </DatePicker.Control>
  <DatePicker.Positioner>
    <DatePicker.Content>
      <DatePicker.View view="day">
        <DatePicker.ViewControl>
          <DatePicker.PrevTrigger />
          <DatePicker.MonthSelect />
          <DatePicker.YearSelect />
          <DatePicker.NextTrigger />
        </DatePicker.ViewControl>
        {/* ... table */}
      </DatePicker.View>
    </DatePicker.Content>
  </DatePicker.Positioner>
</DatePicker.Root>
```

This avoids the navigation inconsistencies with the view switcher in multi-month layouts.
